src/components/dropdown-filter-ajax/dropdown-filter-ajax.js

```diff
@@ -207,26 +225,39 @@ class DropdownFilterAjax extends DropdownFilter {
    *
    * @method getData
    * @param {String} query The search term
-   * @param {Object} page The page number to get
    */
-  getData = (query = '', page = 1) => {
+  getData = (query = '') => {
     this.setState({ requesting: true });
     Request
       .get(this.props.path)
-      .query({
-        page,
-        rows: this.props.rowsPerRequest,
-        value: query
-      })
+      .query(this.getParams(query))
       .query(this.props.additionalRequestParams)
       .end(this.ajaxUpdateList);
   }
 
+  /**
+   * Retrieve params for the list.
+   *
+   * @method getParams
+   */
+  getParams = (query) => {
+    const params = {};
+    params.page = this.state.page;
+    params.rows = this.props.rowsPerRequest;
+    params.value = query;
+    if (this.props.formatRequest) {
+      return this.props.formatRequest(params);
+    }
+    return params;
+  }
+
   /**
    * Applies some data from AJAX to the list
    */
   ajaxUpdateList = (err, response) => {
-    this.updateList(response.body.data[0]);
+    this.updateList(
+      this.props.formatResponse ? this.props.formatResponse(response.body.data) : response.body.data[0]
```

Comment:
It has to be `response.body` as `data` already goes into the JSON response structure and makes assumption about what the response looks like. So while we keep it `body.data[0]` for old path, we need just `body` when we use formatting callback.